### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by swenson**
By running:

```sh
go list -u -m -json all | jq -r 'select(.Indirect != true and .Update != null) | .Path+"@"+.Update.Version' | xargs -L1 go get
go mod tidy
```

|             MODULE              | VERSION | NEW VERSION | DIRECT | VALID TIMESTAMPS |
|---------------------------------|---------|-------------|--------|------------------|
| github.com/cenkalti/backoff/v3  | v3.0.0  | v3.2.2      | true   | true             |
| github.com/hashicorp/go-hclog   | v0.16.2 | v1.3.1      | true   | true             |
| github.com/hashicorp/go-version | v1.2.0  | v1.6.0      | true   | true             |
| github.com/hashicorp/vault/api  | v1.7.2  | v1.8.2      | true   | true             |
| github.com/hashicorp/vault/sdk  | v0.5.3  | v0.6.1      | true   | true             |
| github.com/stretchr/testify     | v1.8.0  | v1.8.1      | true   | true             |
| gopkg.in/square/go-jose.v2      | v2.5.1  | v2.6.0      | true   | true             |
| k8s.io/api                      | v0.22.2 | v0.25.3     | true   | true             |
| k8s.io/apimachinery             | v0.22.2 | v0.25.3     | true   | true             |
| k8s.io/client-go                | v0.22.2 | v0.25.3     | true   | true             |

I ran the integration tests locally, and they passed.

I also checked that the Vault tests pass after updating this plugin (and the k8s secrets plugin): https://github.com/hashicorp/vault/commit/ce44aac88b6852b711fdc18b8ca39ee5f97b2d69

# Related Issues/Pull Requests

- [ ] https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/166

# Contributor Checklist

- [x] Add relevant docs to upstream Vault repository, or sufficient reasoning why docs won’t be added yet (N/A)
- [x] Add output for any tests not ran in CI to the PR description (eg, acceptance tests)
- [x] Backwards compatible

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+| github.com/cenkalti/backoff/v3  | v3.0.0  | v3.2.2      | true   | true             |
+| github.com/hashicorp/go-hclog   | v0.16.2 | v1.3.1      | true   | true             |
+| github.com/hashicorp/go-version | v1.2.0  | v1.6.0      | true   | true             |
+| github.com/hashicorp/vault/api  | v1.7.2  | v1.8.2      | true   | true             |
+| github.com/hashicorp/vault/sdk  | v0.5.3  | v0.6.1      | true   | true             |
+| github.com/stretchr/testify     | v1.8.0  | v1.8.1      | true   | true             |
+| gopkg.in/square/go-jose.v2      | v2.5.1  | v2.6.0      | true   | true             |
+| k8s.io/api                      | v0.22.2 | v0.25.3     | true   | true             |
+| k8s.io/apimachinery             | v0.22.2 | v0.25.3     | true   | true             |
+| k8s.io/client-go                | v0.22.2 | v0.25.3     | true   | true             |
```
Comment (tvoran) [inline CHANGELOG.md:16]: Should these be in a Changes or Improvements section? And I'm curious what "valid timestamps" means?

**Review by tvoran — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (tomhjp) [inline CHANGELOG.md:16]: How come these two are in the second `require` block but are direct dependencies? Doesn't really matter though

**Review by tomhjp — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (swenson) [reply CHANGELOG.md:16]: I think it means if the version has a timestamp and commit in it, like 20220105..., that the timestamp is valid?

I'll add it to the changes section I think.

Comment (swenson) [reply CHANGELOG.md:16]: Not sure. I'll run `go mod tidy` again just in case.

Comment (swenson) [reply CHANGELOG.md:18]: `go mod tidy` didn't fix it, but I manually moved them up and ran `go mod tidy` again, and that seemed to do the trick. Weird.

**Comment by swenson** [comment]
Thanks!

**Pull request merged**